### PR TITLE
Prevent levelbuilders from accidentally switching a standalone course to non-standalone

### PR DESCRIPTION
**Pull request by lfryemason**
Yesterday, a standalone course unit (20-hour) was added to a course. It was immediately removed, but the side-effects were that the standalone course was marked as not standalone. This course was used in many tests and no longer showed the same properties, causing test builds to fail and DTP to be delayed until resolution.

The fix for the unit was to revert the changes to the course, however we want to prevent this from happening in the future.

This PR prevents the names of standalone courses from being included in the dropdown to add to other courses. This should prevent the above error from happening in the future.

This PR also updates the language to make it clearer on unit creation what the options for standalone and non-standalone courses do.

## Links
Slack discussion: https://codedotorg.slack.com/archives/C045UAX4WKH/p1702490190936009
[TEACH-774](https://codedotorg.atlassian.net/browse/TEACH-774)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/controllers/courses_controller.rb [hunk 1]
```diff
@@ -98,7 +98,7 @@ def edit
     raise ActiveRecord::ReadOnlyRecord if @unit_group.try(:plc_course)
     @unit_group_data = {
       course_summary: @unit_group.summarize(@current_user, for_edit: true),
-      script_names: Unit.all.map(&:name),
+      script_names: Unit.all.select {|unit| unit.is_course? == false}.map(&:name),
```
Comment (lfryemason) [inline dashboard/app/controllers/courses_controller.rb:101]: This is the main change, it removes all units that are standalone courses from showing up in the dropdown.

**Review by kobryan0619 — APPROVED** [review]
Looks good to me!

File: dashboard/app/controllers/courses_controller.rb (hunk 1 above)
Comment (davidsbailey) [reply dashboard/app/controllers/courses_controller.rb:101]: I'm a little confused why this method exists alongside the `is_course` serialized property and which one is the source of truth, but for this kind of change I think this is good enough

**Review by davidsbailey — APPROVED** [review]
Thanks for tackling this Liam!

**Comment by lfryemason** [comment]
I think it is because this method is on the Unit and `is_course` is on the UnitGroup

**Comment by davidsbailey** [comment]
is_course is on the unit also: https://github.com/code-dot-org/code-dot-org/blob/ef9c233e40e4df19f1edab1221f7fd6626f33b85/dashboard/app/models/unit.rb#L311

**Pull request merged**